modules/framework/galasa-parent/dev.galasa.framework.api.resources/src/test/java/dev/galasa/framework/api/resources/processors/GalasaTagProcessorTest.java

```diff
@@ -0,0 +1,346 @@
+        // Given...
+        MockRBACService mockRbacService = FilledMockRBACService.createTestRBACServiceWithTestUser(JWT_USERNAME);
+
+        MockTagsService tagService = new MockTagsService();
+
+        RBACValidator rbacValidator = new RBACValidator(mockRbacService);
+        GalasaTagProcessor tagProcessor = new GalasaTagProcessor(tagService, rbacValidator);
+
+        String tagName = "     ";
+        String description = "This is a test tag";
+        Integer priority = 100;
+        String requestUsername = "myuser";
+
+        JsonObject tagJson = generateTagJson(tagName, description, priority);
+
+        // When...
+        List<String> errors = tagProcessor.processResource(tagJson, DELETE, requestUsername);
+
+        // Then...
+        assertThat(errors).isNotEmpty();
+        assertThat(errors.get(0)).contains("GAL5443E", "Invalid tag name provided");
+    }
+
+    @Test
+    public void testCanDeleteTag() throws Exception {
+        // Given...
+        String tagName = "mytag";
+        String description = "This is a test tag";
+        Integer priority = 100;
+        String requestUsername = "myuser";
+
+        MockRBACService mockRbacService = FilledMockRBACService.createTestRBACServiceWithTestUser(JWT_USERNAME);
+
+        Tag tag = new Tag(tagName);
+        tag.setName(tagName);
+        tag.setDescription(description);
+        tag.setPriority(priority);
+
+        Map<String, Tag> tags = new HashMap<>();
+        tags.put(tag.getName(), tag);
+
+        MockTagsService tagService = new MockTagsService(tags);
+
+        RBACValidator rbacValidator = new RBACValidator(mockRbacService);
+        GalasaTagProcessor tagProcessor = new GalasaTagProcessor(tagService, rbacValidator);
+
+        JsonObject tagJson = generateTagJson(tagName, description, priority);
+
+        // Check that we have a tag before processing
+        assertThat(tags).hasSize(1);
+
+        // When...
+        List<String> errors = tagProcessor.processResource(tagJson, DELETE, requestUsername);
+
+        // Then...
+        assertThat(errors).isEmpty();
+        assertThat(tags).isEmpty();
+    }
+
+    @Test
+    public void testCanCreateTag() throws Exception {
+        // Given...
+        String tagName = "mytag";
+        String description = "This is a test tag";
+        Integer priority = 100;
+        String requestUsername = "myuser";
+
+        MockRBACService mockRbacService = FilledMockRBACService.createTestRBACServiceWithTestUser(JWT_USERNAME);
+
+        Map<String, Tag> tags = new HashMap<>();
+
+        MockTagsService tagService = new MockTagsService(tags);
+
+        RBACValidator rbacValidator = new RBACValidator(mockRbacService);
+        GalasaTagProcessor tagProcessor = new GalasaTagProcessor(tagService, rbacValidator);
+
+        JsonObject tagJson = generateTagJson(tagName, description, priority);
+
+        // Check that we have a tag before processing
```

Comment:
Minor - `that we have no existing tags` or something

Reply:
Great spot! Fixed in https://github.com/galasa-dev/galasa/pull/481/commits/b670b34cd3936d79c5176e89de2f6eecb6385bab